modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IncrementalSnapshotVerificationTask.java

```diff
@@ -0,0 +1,435 @@
+                    }
+
+                    @Override void processedWalSegments(int segCnt) {
+                        // No-op.
+                    }
+
+                    @Override void initWalEntries(LongAdder entriesCnt) {
+                        // No-op.
+                    }
+                };
+
+                short locShortId = blt.consistentIdMapping().get(consId);
+
+                Set<GridCacheVersion> activeDhtTxs = new HashSet<>();
+                Map<GridCacheVersion, Set<Short>> txPrimPartNodes = new HashMap<>();
+                Map<Short, PartitionHashHolder> nodesTxHash = new HashMap<>();
+
+                Set<GridCacheVersion> partialCommittedTxs = new HashSet<>();
+                Map<PartitionKeyV2, PartitionHashHolder> partMap = new HashMap<>();
+                List<Exception> exceptions = new ArrayList<>();
+
+                BiConsumer<GridCacheVersion, Set<Short>> calcTransactionHash = (xid, partNodes) -> {
+                    for (short shortId: partNodes) {
+                        if (shortId != locShortId) {
+                            PartitionHashHolder hash = nodesTxHash.computeIfAbsent(shortId, (k) -> new PartitionHashHolder());
+
+                            hash.increment(xid.hashCode(), 0);
+                        }
+                    }
+                };
+
+                // CacheId -> CacheGrpId.
+                Map<Integer, Integer> cacheGrpId = txCaches.values().stream()
+                    .collect(Collectors.toMap(
+                        StoredCacheData::cacheId,
+                        cacheData -> CU.cacheGroupId(cacheData.config().getName(), cacheData.config().getGroupName())
+                    ));
+
+                proc.process(dataEntry -> {
+                    if (dataEntry.op() == GridCacheOperation.READ)
+                        return;
+
+                    if (log.isTraceEnabled())
+                        log.trace("Checking data entry [entry=" + dataEntry + ']');
+
+                    if (!activeDhtTxs.contains(dataEntry.writeVersion()))
+                        partialCommittedTxs.add(dataEntry.nearXidVersion());
+
+                    StoredCacheData cacheData = txCaches.get(dataEntry.cacheId());
+
+                    PartitionKeyV2 partKey = new PartitionKeyV2(
+                        cacheGrpId.get(dataEntry.cacheId()),
+                        dataEntry.partitionId(),
+                        CU.cacheOrGroupName(cacheData.config()));
+
+                    PartitionHashHolder hash = partMap.computeIfAbsent(partKey, (k) -> new PartitionHashHolder());
+
+                    try {
+                        int valHash = dataEntry.key().hashCode() + Arrays.hashCode(dataEntry.value().valueBytes(null));
+                        int verHash = dataEntry.writeVersion().hashCode();
+
+                        hash.increment(valHash, verHash);
+                    }
+                    catch (IgniteCheckedException ex) {
+                        exceptions.add(ex);
```

Comment:
Do we really want to continue task in case of any error?
Looks like we can just skip all further actions like:

```
                proc.process(dataEntry -> {
                    if (dataEntry.op() == GridCacheOperation.READ || !exceptions.isEmpty())
                        return;

...

                }, txRec -> {
                    if (!exceptions.isEmpty())
                         return;

                    if (log.isDebugEnabled())
                        log.debug("Checking tx record [txRec=" + txRec + ']');
```